### ISSUE TITLE
If a censored word had punctuation following it then it is removed when asterisked/censored

### ISSUE THREAD

**Issue by HammyHavoc** [RocketChat/Rocket.Chat#8824]
### Description:

When the blacklist of words to censor is enabled, any word that is censored which had punctuation following it ends up losing the punctuation following the asterisks. So if I send a message asking someone "Please could you bring over the knobs? This guitar is almost done." then the 'knobs' is censored (silly, I know), but the ***** loses the ? following it, which makes the flow of conversation hard to follow.

**Comment by subham103** [comment]
I would like to work on this issue:smile:

**Comment by subham103** [comment]
Can anybody tell me how can we censor/asterisk a word?

**Comment by subham103** [comment]
@HammyHavoc ^

**Comment by HammyHavoc** [comment]
At this present time, I do not use Rocket.Chat, nor contribute to the development of it. We had to find another solution for our legally blind colleague due to a lack of dark mode and no progress made on that front when we were forced to switch to another solution.

**Comment by subham103** [comment]
@ggazzo, can you please tell me how we can censor/asterisk a word, I am facing difficulty in it.

**Comment by ashwaniYDV** [comment]
@HammyHavoc @ggazzo 
Is this issue still valid?

I am not able to produce it.

![11](https://user-images.githubusercontent.com/43786728/74946642-9e9c8a80-541f-11ea-9ad1-85173f1aad58.gif)

**Comment by ashwaniYDV** [comment]
> Can anybody tell me how can we censor/asterisk a word?

You can do so by going `administraion->messages`

**Comment by subham103** [comment]
> @HammyHavoc @ggazzo @gabriellsh
> Is this issue still valid?
> 
> I am not able to produce it.
> knobs is by default a badword in rocket.chat
> I have added custom badword `ashwani` too and verified.
> 
> ![11](https://user-images.githubusercontent.com/43786728/74946642-9e9c8a80-541f-11ea-9ad1-85173f1aad58.gif)

same here. I am also unable to reproduce this issue. Instead, the problem is when we put a word between two censored words, we unintendedly made that word bold, This issue is opened  #16670

**Comment by ashwaniYDV** [comment]
@subham103 i have made a pr for the "bold" issue

**Comment by subham103** [comment]
@ashwaniYDV The bad word filtering is already conditional. There is a toggle option (Allow Message bad words filtering) already exist in `administraion->messages` which upon off will not change the bad words to ****

**Comment by ashwaniYDV** [comment]
@subham103 The issue here is that if bad words filtering is on and I enter a message `knob is a bad word`, then it is saved in the database as `**** is a bad word`. Now if I turn off the bad word filter from settings, then I expect the message to be `knob is a bad word`.

**Comment by gabriellsh** [comment]
I'll be closing this one as this specific issue doesn't happen. Please, take the discussion to #16670.
